projects/storefrontlib/layout/main/storefront.component.ts

```diff
@@ -138,4 +143,29 @@ export class StorefrontComponent implements OnInit, OnDestroy {
       this.navigateSubscription.unsubscribe();
     }
   }
+
+  protected onNavigation(isNavigating: boolean): void {
+    this.startNavigating = isNavigating === true;
+    this.stopNavigating = isNavigating === false;
+    if (
+      this.featureConfigService.isEnabled('a11yResetFocusAfterNavigating') &&
+      this.stopNavigating &&
+      // we only need to reset focus after clicking a link
+      this.document?.activeElement !== this.document?.body
+    ) {
+      this.skipFocusTo('cx-main');
+    }
+  }
+
+  protected skipFocusTo(skipLinkKey: string): void {
+    this.skipLinkService
+      ?.getSkipLinks()
+      .pipe(take(1))
+      .subscribe((skipLinks) => {
+        const skipLink = skipLinks.find((link) => link.key === skipLinkKey);
+        if (skipLink) {
+          this.skipLinkService?.scrollToTarget(skipLink);
+        }
+      });
```

Comment:
Interesting solution to use the skip link service.

Maybe we could have `key` as a possible parameter for the `scrollToTarget` method OR `scrollToTargetByKey` in the skip link service as this method could be useful in other places. I don't think we need `skipFocusTo` in the storefront component.

Reply:
Good suggestion!